src/bin/cargo-info/command/info.rs

```diff
@@ -102,6 +102,13 @@ pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
         .map(String::as_str)
         .unwrap();
     let spec = PackageIdSpec::parse(package)?;
+    if spec.name().is_empty() {
+        return Err(CliError::new(
```

Comment:
Copied from cargo. So I guess it is the correct way to bail out an error with the error code.